jmix-flowui/flowui/src/main/java/io/jmix/flowui/component/genericfilter/FilterMetadataTools.java

```diff
@@ -53,24 +54,32 @@ public FilterMetadataTools(MetadataTools metadataTools,
         this.metadata = metadata;
     }
 
-    public List<MetaPropertyPath> getPropertyPaths(MetaClass filterMetaClass,
-                                                   String query,
-                                                   @Nullable Predicate<MetaPropertyPath> propertiesFilterPredicate) {
-        return getPropertyPaths(filterMetaClass, query, filterMetaClass, 0, "",
-                propertiesFilterPredicate);
+    public List<MetaPropertyPath> getPropertyPaths(GenericFilter filter) {
```

Comment:
We don't change public API without a real need. In case of `propertyHierarchyDepth`:
1. `GenericFilter` component can load default value from properties, so it will return a value. 
2. If you need to get actual value from a filter, create a new method that accepts this value. The old method should pass a value from properties as a default value.